packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_loan_update_vault.test.ts

```diff
@@ -0,0 +1,295 @@
+    })
+    await alice.generate(1)
+
+    // Loan token
+    await alice.rpc.loan.setLoanToken({
+      symbol: 'TSLA',
+      fixedIntervalPriceId: 'TSLA/USD'
+    })
+    await alice.generate(1)
+
+    // Create a default vault
+    vaultId = await alice.rpc.loan.createVault({
+      ownerAddress: collateralAddress,
+      loanSchemeId: 'default'
+    })
+    await alice.generate(1)
+  })
+
+  afterAll(async () => {
+    await tGroup.stop()
+  })
+
+  beforeEach(async () => {
+    // Always update the vault back to default scheme
+    await alice.rpc.loan.updateVault(vaultId, {
+      ownerAddress: collateralAddress,
+      loanSchemeId: 'default'
+    })
+    await alice.generate(1)
+
+    const data = await alice.rpc.loan.getVault(vaultId)
+    expect(data.loanSchemeId).toStrictEqual('default')
+
+    // Fund 10 DFI UTXO
+    await fundEllipticPair(alice.container, providers.ellipticPair, 10)
+    await providers.setupMocks() // Required to move utxos
+  })
+
+  it('should updateVault', async () => {
+    const script = await providers.elliptic.script()
+    const txn = await builder.loans.updateVault({
+      vaultId,
+      ownerAddress: script,
+      schemeId: 'scheme'
+    }, script)
+
+    // Ensure the created txn is correct
+    const outs = await sendTransaction(alice.container, txn)
+    expect(outs[0].value).toStrictEqual(0)
+    expect(outs[1].value).toBeLessThan(10)
+    expect(outs[1].value).toBeGreaterThan(9.999)
+    expect(outs[1].scriptPubKey.addresses[0]).toStrictEqual(await providers.getAddress())
+
+    // Ensure you don't send all your balance away
+    const prevouts = await providers.prevout.all()
+    expect(prevouts.length).toStrictEqual(1)
+    expect(prevouts[0].value.toNumber()).toBeLessThan(10)
+
+    const data = await alice.rpc.loan.getVault(vaultId)
+    expect(data).toStrictEqual({
+      vaultId,
+      loanSchemeId: 'scheme',
+      ownerAddress: collateralAddress,
+      state: 'active',
+      collateralAmounts: [],
+      loanAmounts: [],
+      interestAmounts: [],
+      collateralValue: expect.any(BigNumber),
+      loanValue: expect.any(BigNumber),
+      interestValue: new BigNumber(0),
+      collateralRatio: -1,
+      informativeRatio: new BigNumber(-1)
+    })
+  })
+
+  it('should updateVault if the vault state is mayliquidate', async () => {
+    // Create another vault
+    const vaultId = await alice.rpc.loan.createVault({
+      ownerAddress: collateralAddress,
+      loanSchemeId: 'default'
+    })
+    await alice.generate(1)
+
+    // Deposit to vault
+    await alice.rpc.loan.depositToVault({
+      vaultId, from: collateralAddress, amount: '100@DFI'
+    })
+    await alice.generate(1)
+
+    // Take loan
+    await alice.rpc.loan.takeLoan({
+      vaultId,
+      amounts: '50@TSLA'
+    })
+    await alice.generate(1)
+
+    await alice.rpc.oracle.setOracleData(oracleId, Math.floor(new Date().getTime() / 1000), { prices: [{ tokenAmount: '0.9@DFI', currency: 'USD' }] })
```

Comment:
generate a block after this? coz of https://github.com/DeFiCh/jellyfish/pull/816 ?
```suggestion
    await alice.rpc.oracle.setOracleData(oracleId, Math.floor(new Date().getTime() / 1000), { prices: [{ tokenAmount: '0.9@DFI', currency: 'USD' }] })
    await alice.generate(1)
```

Reply:
Will do. You are too fast.